src/vmspawn/vmspawn.c

```diff
@@ -242,6 +247,31 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
+                case ARG_FIRMWARE:
+                        if (streq(optarg, "list")) {
+                                _cleanup_strv_free_ char **l = NULL;
+
+                                r = list_ovmf_config(&l);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to list firmwares: %m");
+
+                                bool nl = false;
+                                fputstrv(stdout, l, "\n", &nl);
+                                if (nl)
+                                        putchar('\n');
+
+                                return 0;
+                        }
+
+                        if (!isempty(optarg) && !path_is_absolute(optarg) && !startswith(optarg, "./"))
```

Comment:
Should we allow ../ here?

Reply:
The only reason why I insist on the ./ here is that i wanted to avoid the ambiguity if "--firmware=auto" means "automatically find the firmware" or "load firmware from file 'auto' in cwd".

By insisting on ./ this ambiguities go away: by looking at the first characters it's always clear what this is: "./" → path relative to cwd, "/" → absolute path, anything else is special string. This nicely keeps the door for future special strings.

Now, we could add "../" here as well, but i think it just would readd redundance, since it's already covered by "./../" after all. or to say this differently, everyone knows absolute and relative paths, but noone thinks of "relative paths to parent of cwd" as a basic concept we must say.

or to say it even differently: it would make the docs unnecessarily long, as we'd have to cover this case too.